teamcity: failed tests on master: testrace/TestRefreshPendingCommands

The following tests appear to have failed:

[#84831](https://teamcity.cockroachdb.com/viewLog.html?buildId=84831):

```
--- FAIL: testrace/TestRefreshPendingCommands (0.014s)
Race detected!
I161208 20:54:12.321343 12868 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:12.322393 12868 gossip/gossip.go:248  [n?] initial resolvers: []
W161208 20:54:12.322500 12868 gossip/gossip.go:1124  [n?] no resolvers found; use --join to specify a connected node
I161208 20:54:12.322656 12868 base/node_id.go:62  NodeID set to 1
I161208 20:54:12.331579 12868 storage/store.go:1232  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161208 20:54:12.333575 12868 gossip/gossip.go:290  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:57135" > attrs:<> locality:<> 
I161208 20:54:12.339446 12860 storage/replica_proposal.go:349  [s1,r1/1:/M{in-ax},@c420212000] new range lease repl={1 1 1} start=0.000000000,0 exp=0.900000123,16 pro=0.000000123,17 following repl={0 0 0} start=0.000000000,0 exp=0.000000000,0 [physicalTime=1970-01-01 00:00:00.000000123 +0000 UTC]
I161208 20:54:12.344245 12868 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:12.345226 12868 gossip/gossip.go:248  [n?] initial resolvers: [127.0.0.1:57135]
W161208 20:54:12.345307 12868 gossip/gossip.go:1126  [n?] no incoming or outgoing connections
I161208 20:54:12.345450 12868 base/node_id.go:62  NodeID set to 2
I161208 20:54:12.349559 12930 gossip/client.go:125  [n2] started gossip client to 127.0.0.1:57135
I161208 20:54:12.351029 12868 storage/store.go:1232  [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I161208 20:54:12.351218 12868 gossip/gossip.go:290  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:46266" > attrs:<> locality:<> 
I161208 20:54:12.421206 12868 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:12.422264 12868 gossip/gossip.go:248  [n?] initial resolvers: [127.0.0.1:57135]
W161208 20:54:12.422348 12868 gossip/gossip.go:1126  [n?] no incoming or outgoing connections
I161208 20:54:12.422491 12868 base/node_id.go:62  NodeID set to 3
I161208 20:54:12.428742 12831 gossip/client.go:125  [n3] started gossip client to 127.0.0.1:57135
I161208 20:54:12.429819 12868 storage/store.go:1232  [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I161208 20:54:12.430036 12868 gossip/gossip.go:290  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:37540" > attrs:<> locality:<> 
I161208 20:54:12.444117 12868 storage/replica_raftstorage.go:468  [s1,r1/1:/M{in-ax},@c420212000] generated preemptive snapshot 5152b5c4 at index 15
I161208 20:54:12.446914 13042 storage/replica_raftstorage.go:633  [s2,r1/?:{-},@c420178500] applying preemptive snapshot at index 15 (id=5152b5c4, encoded size=5117, 1 rocksdb batches, 5 log entries)
I161208 20:54:12.447761 13042 storage/replica_raftstorage.go:641  [s2,r1/?:/M{in-ax},@c420178500] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:12.448261 12868 storage/store.go:3229  [s1,r1/1:/M{in-ax},@c420212000] streamed snapshot: kv pairs: 34, log entries: 5
I161208 20:54:12.449999 12868 storage/replica_command.go:3279  [s1,r1/1:/M{in-ax},@c420212000] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I161208 20:54:12.453539 12972 storage/replica.go:2334  [s1,r1/1:/M{in-ax},@c420212000] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I161208 20:54:12.458766 12868 storage/replica_raftstorage.go:468  [s1,r1/1:/M{in-ax},@c420212000] generated preemptive snapshot 1bdaa7ac at index 17
I161208 20:54:12.461657 13045 storage/replica_raftstorage.go:633  [s3,r1/?:{-},@c42020a000] applying preemptive snapshot at index 17 (id=1bdaa7ac, encoded size=6053, 1 rocksdb batches, 7 log entries)
I161208 20:54:12.462568 13045 storage/replica_raftstorage.go:641  [s3,r1/?:/M{in-ax},@c42020a000] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:12.462975 12868 storage/store.go:3229  [s1,r1/1:/M{in-ax},@c420212000] streamed snapshot: kv pairs: 37, log entries: 7
I161208 20:54:12.473630 12868 storage/replica_command.go:3279  [s1,r1/1:/M{in-ax},@c420212000] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I161208 20:54:12.481355 13074 storage/raft_transport.go:437  raft transport stream to node 1 established
I161208 20:54:12.492984 12934 storage/replica.go:2334  [s1,r1/1:/M{in-ax},@c420212000] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I161208 20:54:12.551569 12868 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
W161208 20:54:12.552092 13049 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:12.552980 13096 storage/store.go:3091  [s1] raft error: node 3 claims to not contain store 3 for replica {3 3 3}: store 3 was not found
W161208 20:54:12.553124 13094 storage/raft_transport.go:443  raft transport stream to node 3 failed: store 3 was not found
I161208 20:54:12.555228 13098 storage/raft_transport.go:437  raft transport stream to node 3 established
W161208 20:54:12.557217 13055 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:12.557651 13055 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:12.557951 13055 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:12.558099 13100 storage/store.go:3091  [s1] raft error: node 3 claims to not contain store 3 for replica {3 3 3}: store 3 was not found
I161208 20:54:12.562175 13101 storage/raft_transport.go:437  raft transport stream to node 3 established
W161208 20:54:12.562624 13107 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:12.563034 13057 storage/store.go:3091  [s1] raft error: node 3 claims to not contain store 3 for replica {3 3 3}: store 3 was not found
I161208 20:54:12.563955 12868 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
E161208 20:54:12.577430 13079 storage/client_test.go:1108  result is ambiguous (may have succeeded at failing replica)
E161208 20:54:12.585385 13079 storage/client_test.go:1108  result is ambiguous (may have succeeded at failing replica)
==================
WARNING: DATA RACE
Read at 0x00c420010e70 by goroutine 76:
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).expireLeases()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:1102 +0x148
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContextKVTransport).SendNext.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:495 +0x626
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:264 +0xec

Previous write at 0x00c420010e70 by goroutine 197:
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).restartStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:867 +0x426
  github.com/cockroachdb/cockroach/pkg/storage_test.TestRefreshPendingCommands.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:986 +0xb22
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Goroutine 76 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:265 +0x2af
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContextKVTransport).SendNext()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:500 +0x51f
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1142 +0x375
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:410 +0x417
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:479 +0x1aa
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendPartialBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:931 +0x3b3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:810 +0x567
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:625 +0x381
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:418 +0x7f1
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).sendInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:587 +0x16e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:711 +0x63c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).(github.com/cockroachdb/cockroach/pkg/internal/client.send)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:313 +0x7c
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:418 +0x1ab
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:313 +0xfd
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:356 +0x55b
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:468 +0x46
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:520 +0x233
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:469 +0x297
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:357 +0x139
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).Heartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:217 +0x2d3
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:163 +0x74c
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Goroutine 197 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:646 +0x52f
  github.com/cockroachdb/cockroach/pkg/storage_test.TestRefreshPendingCommands()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:997 +0x226
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9
==================
I161208 20:54:12.951989 13089 storage/raft_transport.go:437  raft transport stream to node 3 established
I161208 20:54:12.955304 12896 storage/replica_raftstorage.go:468  [s2,r1/2:/M{in-ax},@c420178500] generated Raft snapshot e31c9890 at index 24
I161208 20:54:12.958922 12941 storage/store.go:3229  [s2,r1/2:/M{in-ax},@c420178500] streamed snapshot: kv pairs: 44, log entries: 1
I161208 20:54:12.959244 13172 storage/replica_raftstorage.go:633  [s3,r1/3:/M{in-ax},@c42020b400] applying Raft snapshot at index 24 (id=e31c9890, encoded size=4336, 1 rocksdb batches, 1 log entries)
I161208 20:54:12.960349 13172 storage/replica_raftstorage.go:641  [s3,r1/3:/M{in-ax},@c42020b400] applied Raft snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:12.964118 12952 storage/replica_proposal.go:405  [s2,r1/2:/M{in-ax},@c420178500] range [n2,s2,r1/2:/M{in-ax}]: transferring raft leadership to replica ID 3
I161208 20:54:12.965986 13159 storage/replica_proposal.go:349  [s3,r1/3:/M{in-ax},@c42020b400] new range lease repl={3 3 3} start=0.900000123,17 exp=4.500000127,48 pro=3.600000127,49 following repl={1 1 1} start=0.000000000,0 exp=0.900000123,16 pro=0.000000123,17 [physicalTime=1970-01-01 00:00:03.600000127 +0000 UTC]
E161208 20:54:13.026748 13136 storage/node_liveness.go:167  [hb] failed liveness heartbeat: context deadline exceeded
I161208 20:54:13.144003 13253 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.144953 13253 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.145876 13253 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.146630 13253 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
W161208 20:54:13.148858 13074 storage/raft_transport.go:443  raft transport stream to node 1 failed: rpc error: code = 13 desc = transport is closing
I161208 20:54:13.149203 12848 vendor/google.golang.org/grpc/transport/http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:46266->127.0.0.1:53036: use of closed network connection
I161208 20:54:13.149444 12885 vendor/google.golang.org/grpc/transport/http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I161208 20:54:13.150032 12887 vendor/google.golang.org/grpc/clientconn.go:667  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:46266: operation was canceled"; Reconnecting to {"127.0.0.1:46266" <nil>}
I161208 20:54:13.150150 12887 vendor/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: grpc: the connection is closing
I161208 20:54:13.150429 13253 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.150725 13253 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.150916 13253 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.151604 13254 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:13.152629 13254 gossip/gossip.go:248  [n?] initial resolvers: []
W161208 20:54:13.152715 13254 gossip/gossip.go:1124  [n?] no resolvers found; use --join to specify a connected node
I161208 20:54:13.152888 13254 base/node_id.go:62  NodeID set to 1
I161208 20:54:13.162012 13254 storage/store.go:1232  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161208 20:54:13.162206 13254 gossip/gossip.go:290  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46334" > attrs:<> locality:<> 
I161208 20:54:13.169423 13268 storage/replica_proposal.go:349  [s1,r1/1:/M{in-ax},@c420312500] new range lease repl={1 1 1} start=0.000000000,0 exp=0.900000123,16 pro=0.000000123,17 following repl={0 0 0} start=0.000000000,0 exp=0.000000000,0 [physicalTime=1970-01-01 00:00:00.000000123 +0000 UTC]
I161208 20:54:13.181015 13254 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:13.182102 13254 gossip/gossip.go:248  [n?] initial resolvers: [127.0.0.1:46334]
W161208 20:54:13.182199 13254 gossip/gossip.go:1126  [n?] no incoming or outgoing connections
I161208 20:54:13.182356 13254 base/node_id.go:62  NodeID set to 2
I161208 20:54:13.187806 13294 gossip/client.go:125  [n2] started gossip client to 127.0.0.1:46334
I161208 20:54:13.192360 13254 gossip/gossip.go:290  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:39883" > attrs:<> locality:<> 
I161208 20:54:13.205965 13254 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:13.207038 13254 gossip/gossip.go:248  [n?] initial resolvers: [127.0.0.1:46334]
W161208 20:54:13.207127 13254 gossip/gossip.go:1126  [n?] no incoming or outgoing connections
I161208 20:54:13.207270 13254 base/node_id.go:62  NodeID set to 3
I161208 20:54:13.211288 13214 gossip/client.go:125  [n3] started gossip client to 127.0.0.1:46334
I161208 20:54:13.213610 13254 storage/store.go:1232  [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I161208 20:54:13.213835 13254 gossip/gossip.go:290  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:55377" > attrs:<> locality:<> 
I161208 20:54:13.223382 13254 storage/replica_raftstorage.go:468  [s1,r1/1:/M{in-ax},@c420312500] generated preemptive snapshot 0d287732 at index 15
I161208 20:54:13.225387 13254 storage/store.go:3229  [s1,r1/1:/M{in-ax},@c420312500] streamed snapshot: kv pairs: 34, log entries: 5
I161208 20:54:13.226156 13443 storage/replica_raftstorage.go:633  [s2,r1/?:{-},@c42091e000] applying preemptive snapshot at index 15 (id=0d287732, encoded size=5117, 1 rocksdb batches, 5 log entries)
I161208 20:54:13.227086 13443 storage/replica_raftstorage.go:641  [s2,r1/?:/M{in-ax},@c42091e000] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:13.229221 13254 storage/replica_command.go:3279  [s1,r1/1:/M{in-ax},@c420312500] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I161208 20:54:13.233107 13430 storage/replica.go:2334  [s1,r1/1:/M{in-ax},@c420312500] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I161208 20:54:13.238154 13254 storage/replica_raftstorage.go:468  [s1,r1/1:/M{in-ax},@c420312500] generated preemptive snapshot a42c5393 at index 17
I161208 20:54:13.239973 13254 storage/store.go:3229  [s1,r1/1:/M{in-ax},@c420312500] streamed snapshot: kv pairs: 37, log entries: 7
I161208 20:54:13.242624 13434 storage/replica_raftstorage.go:633  [s3,r1/?:{-},@c420213400] applying preemptive snapshot at index 17 (id=a42c5393, encoded size=6052, 1 rocksdb batches, 7 log entries)
I161208 20:54:13.243540 13434 storage/replica_raftstorage.go:641  [s3,r1/?:/M{in-ax},@c420213400] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:13.247038 13254 storage/replica_command.go:3279  [s1,r1/1:/M{in-ax},@c420312500] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I161208 20:54:13.251283 13225 storage/raft_transport.go:437  raft transport stream to node 1 established
I161208 20:54:13.257556 13459 storage/replica.go:2334  [s1,r1/1:/M{in-ax},@c420312500] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I161208 20:54:13.338757 13254 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
W161208 20:54:13.340943 13230 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:13.342188 13446 storage/store.go:3091  [s1] raft error: node 3 claims to not contain store 3 for replica {3 3 3}: store 3 was not found
W161208 20:54:13.343457 13230 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:13.343789 13444 storage/raft_transport.go:443  raft transport stream to node 3 failed: store 3 was not found
I161208 20:54:13.347681 13425 storage/raft_transport.go:437  raft transport stream to node 3 established
I161208 20:54:13.349206 13254 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
W161208 20:54:13.349470 13474 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:13.349651 13440 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:2 StoreID:2 ReplicaID:2}: no handler registered for {NodeID:1 StoreID:1 ReplicaID:1}
W161208 20:54:13.350184 13227 storage/store.go:3091  [s2] raft error: node 1 claims to not contain store 1 for replica {1 1 1}: store 1 was not found
W161208 20:54:13.350706 13225 storage/raft_transport.go:443  raft transport stream to node 1 failed: store 1 was not found
W161208 20:54:13.350912 13448 storage/raft_transport.go:478  no handler found for store 1 in response range_id:1 from_replica:<node_id:3 store_id:3 replica_id:3 > to_replica:<node_id:1 store_id:1 replica_id:1 > union:<error:<message:"store 3 was not found" transaction_restart:NONE origin_node:0 detail:<store_not_found:<store_id:3 > > now:<wall_time:0 logical:0 > > > 
E161208 20:54:13.354530 13290 storage/node_liveness.go:167  [hb] failed liveness heartbeat: node unavailable; try another peer
I161208 20:54:13.693483 13477 storage/raft_transport.go:437  raft transport stream to node 1 established
I161208 20:54:13.701510 13330 storage/replica_raftstorage.go:468  [s2,r1/2:/M{in-ax},@c42091e000] generated Raft snapshot 73eca69f at index 23
I161208 20:54:13.703715 13463 storage/store.go:3229  [s2,r1/2:/M{in-ax},@c42091e000] streamed snapshot: kv pairs: 42, log entries: 2
I161208 20:54:13.704048 13572 storage/replica_raftstorage.go:633  [s3,r1/3:/M{in-ax},@c420178a00] applying Raft snapshot at index 23 (id=73eca69f, encoded size=4245, 1 rocksdb batches, 2 log entries)
I161208 20:54:13.704411 13331 storage/replica_proposal.go:405  [s2,r1/2:/M{in-ax},@c42091e000] range [n2,s2,r1/2:/M{in-ax}]: transferring raft leadership to replica ID 3
I161208 20:54:13.705242 13572 storage/replica_raftstorage.go:641  [s3,r1/3:/M{in-ax},@c420178a00] applied Raft snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:13.707982 13527 storage/replica_proposal.go:349  [s3,r1/3:/M{in-ax},@c420178a00] new range lease repl={3 3 3} start=0.900000123,17 exp=4.500000127,12 pro=3.600000127,13 following repl={1 1 1} start=0.000000000,0 exp=0.900000123,16 pro=0.000000123,17 [physicalTime=1970-01-01 00:00:03.600000127 +0000 UTC]
E161208 20:54:13.770749 13327 storage/client_test.go:1108  mismatch incrementing epoch for node_id:2 epoch:1 expiration:<wall_time:900000123 logical:37 > ; actual is {NodeID:2 Epoch:1 Expiration:4.500000127,83}
E161208 20:54:13.770957 13327 storage/client_test.go:1108  cannot increment epoch on live node: node_id:3 epoch:1 expiration:<wall_time:4500000127 logical:119 > 
E161208 20:54:13.776046 13398 storage/node_liveness.go:167  [hb] failed liveness heartbeat: context deadline exceeded
I161208 20:54:13.919978 13602 util/stop/stopper.go:468  quiescing; tasks left:
1      storage/client_test.go:500
I161208 20:54:13.921814 13455 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.922690 13455 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.923872 13455 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.929947 13455 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.931814 13263 vendor/google.golang.org/grpc/transport/http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:39883->127.0.0.1:54676: use of closed network connection
I161208 20:54:13.932234 13303 vendor/google.golang.org/grpc/transport/http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I161208 20:54:13.932276 13361 vendor/google.golang.org/grpc/transport/http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:55377->127.0.0.1:48768: use of closed network connection
I161208 20:54:13.932580 13305 vendor/google.golang.org/grpc/clientconn.go:667  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39883: operation was canceled"; Reconnecting to {"127.0.0.1:39883" <nil>}
I161208 20:54:13.932652 13068 vendor/google.golang.org/grpc/transport/http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:46334->127.0.0.1:38223: use of closed network connection
W161208 20:54:13.932939 13222 storage/raft_transport.go:443  raft transport stream to node 2 failed: rpc error: code = 13 desc = transport is closing
I161208 20:54:13.933264 13305 vendor/google.golang.org/grpc/clientconn.go:767  grpc: addrConn.transportMonitor exits due to: grpc: the connection is closing
I161208 20:54:13.933852 13455 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.934148 13455 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.934281 13455 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:13.934964 13374 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:13.935837 13374 gossip/gossip.go:248  [n?] initial resolvers: []
W161208 20:54:13.935928 13374 gossip/gossip.go:1124  [n?] no resolvers found; use --join to specify a connected node
I161208 20:54:13.936073 13374 base/node_id.go:62  NodeID set to 1
I161208 20:54:13.948047 13374 storage/store.go:1232  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161208 20:54:13.948253 13374 gossip/gossip.go:290  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:48706" > attrs:<> locality:<> 
I161208 20:54:13.958711 13638 storage/replica_proposal.go:349  [s1,r1/1:/M{in-ax},@c420178500] new range lease repl={1 1 1} start=0.000000000,0 exp=0.900000123,16 pro=0.000000123,17 following repl={0 0 0} start=0.000000000,0 exp=0.000000000,0 [physicalTime=1970-01-01 00:00:00.000000123 +0000 UTC]
I161208 20:54:13.968756 13374 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:13.969775 13374 gossip/gossip.go:248  [n?] initial resolvers: [127.0.0.1:48706]
W161208 20:54:13.969871 13374 gossip/gossip.go:1126  [n?] no incoming or outgoing connections
I161208 20:54:13.970024 13374 base/node_id.go:62  NodeID set to 2
I161208 20:54:13.974426 13663 gossip/client.go:125  [n2] started gossip client to 127.0.0.1:48706
I161208 20:54:13.975948 13374 storage/store.go:1232  [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I161208 20:54:13.976162 13374 gossip/gossip.go:290  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:53217" > attrs:<> locality:<> 
I161208 20:54:13.986637 13374 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161208 20:54:13.987586 13374 gossip/gossip.go:248  [n?] initial resolvers: [127.0.0.1:48706]
W161208 20:54:13.987678 13374 gossip/gossip.go:1126  [n?] no incoming or outgoing connections
I161208 20:54:13.987831 13374 base/node_id.go:62  NodeID set to 3
I161208 20:54:13.992478 13762 gossip/client.go:125  [n3] started gossip client to 127.0.0.1:48706
I161208 20:54:13.994243 13374 storage/store.go:1232  [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I161208 20:54:13.994416 13374 gossip/gossip.go:290  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:59846" > attrs:<> locality:<> 
I161208 20:54:14.006529 13374 storage/replica_raftstorage.go:468  [s1,r1/1:/M{in-ax},@c420178500] generated preemptive snapshot d2f88242 at index 15
I161208 20:54:14.008802 13374 storage/store.go:3229  [s1,r1/1:/M{in-ax},@c420178500] streamed snapshot: kv pairs: 34, log entries: 5
I161208 20:54:14.009574 13791 storage/replica_raftstorage.go:633  [s2,r1/?:{-},@c420179400] applying preemptive snapshot at index 15 (id=d2f88242, encoded size=5117, 1 rocksdb batches, 5 log entries)
I161208 20:54:14.010477 13791 storage/replica_raftstorage.go:641  [s2,r1/?:/M{in-ax},@c420179400] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:14.013249 13374 storage/replica_command.go:3279  [s1,r1/1:/M{in-ax},@c420178500] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I161208 20:54:14.017153 13810 storage/replica.go:2334  [s1,r1/1:/M{in-ax},@c420178500] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I161208 20:54:14.022811 13374 storage/replica_raftstorage.go:468  [s1,r1/1:/M{in-ax},@c420178500] generated preemptive snapshot 76455925 at index 17
I161208 20:54:14.024420 13374 storage/store.go:3229  [s1,r1/1:/M{in-ax},@c420178500] streamed snapshot: kv pairs: 37, log entries: 7
I161208 20:54:14.031103 13771 storage/replica_raftstorage.go:633  [s3,r1/?:{-},@c420179900] applying preemptive snapshot at index 17 (id=76455925, encoded size=6053, 1 rocksdb batches, 7 log entries)
I161208 20:54:14.032155 13771 storage/replica_raftstorage.go:641  [s3,r1/?:/M{in-ax},@c420179900] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:14.038163 13374 storage/replica_command.go:3279  [s1,r1/1:/M{in-ax},@c420178500] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I161208 20:54:14.041822 13775 storage/raft_transport.go:437  raft transport stream to node 1 established
I161208 20:54:14.046752 13843 storage/replica.go:2334  [s1,r1/1:/M{in-ax},@c420178500] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I161208 20:54:14.201668 13374 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
W161208 20:54:14.204292 13584 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:14.204747 13809 storage/store.go:3091  [s1] raft error: node 3 claims to not contain store 3 for replica {3 3 3}: store 3 was not found
W161208 20:54:14.204872 13807 storage/raft_transport.go:443  raft transport stream to node 3 failed: store 3 was not found
I161208 20:54:14.207410 13852 storage/raft_transport.go:437  raft transport stream to node 3 established
W161208 20:54:14.208371 13633 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:14.208753 13633 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:14.208845 13854 storage/store.go:3091  [s1] raft error: node 3 claims to not contain store 3 for replica {3 3 3}: store 3 was not found
I161208 20:54:14.212741 13831 storage/raft_transport.go:437  raft transport stream to node 3 established
W161208 20:54:14.213276 13614 storage/raft_transport.go:258  unable to accept Raft message from {NodeID:1 StoreID:1 ReplicaID:1}: no handler registered for {NodeID:3 StoreID:3 ReplicaID:3}
W161208 20:54:14.213665 13860 storage/store.go:3091  [s1] raft error: node 3 claims to not contain store 3 for replica {3 3 3}: store 3 was not found
I161208 20:54:14.214074 13374 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:14.576855 13954 storage/raft_transport.go:437  raft transport stream to node 3 established
I161208 20:54:14.579724 13710 storage/replica_raftstorage.go:468  [s2,r1/2:/M{in-ax},@c420179400] generated Raft snapshot d8bad420 at index 25
I161208 20:54:14.582042 13955 storage/store.go:3229  [s2,r1/2:/M{in-ax},@c420179400] streamed snapshot: kv pairs: 45, log entries: 1
I161208 20:54:14.582338 13856 storage/replica_raftstorage.go:633  [s3,r1/3:/M{in-ax},@c420312f00] applying Raft snapshot at index 25 (id=d8bad420, encoded size=4385, 1 rocksdb batches, 1 log entries)
I161208 20:54:14.583459 13856 storage/replica_raftstorage.go:641  [s3,r1/3:/M{in-ax},@c420312f00] applied Raft snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I161208 20:54:14.589357 13713 storage/replica_proposal.go:405  [s2,r1/2:/M{in-ax},@c420179400] range [n2,s2,r1/2:/M{in-ax}]: transferring raft leadership to replica ID 3
I161208 20:54:14.591507 13913 storage/replica_proposal.go:349  [s3,r1/3:/M{in-ax},@c420312f00] new range lease repl={3 3 3} start=0.900000123,17 exp=4.500000127,8 pro=3.600000127,9 following repl={1 1 1} start=0.000000000,0 exp=0.900000123,16 pro=0.000000123,17 [physicalTime=1970-01-01 00:00:03.600000127 +0000 UTC]
E161208 20:54:14.624657 13936 storage/client_test.go:1108  cannot increment epoch on live node: node_id:2 epoch:1 expiration:<wall_time:4500000127 logical:67 > 
E161208 20:54:14.624804 13936 storage/client_test.go:1108  cannot increment epoch on live node: node_id:3 epoch:1 expiration:<wall_time:4500000127 logical:63 > 
E161208 20:54:14.627607 13905 storage/node_liveness.go:167  [hb] failed liveness heartbeat: context deadline exceeded
I161208 20:54:14.782186 13990 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:14.783174 13990 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:14.784159 13990 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:14.785048 13990 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:14.787504 13618 vendor/google.golang.org/grpc/transport/http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:48706->127.0.0.1:46976: use of closed network connection
I161208 20:54:14.787658 13661 vendor/google.golang.org/grpc/transport/http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:53217->127.0.0.1:59665: use of closed network connection
I161208 20:54:14.788307 13990 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:14.788683 13990 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161208 20:54:14.788846 13990 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -1098,8 +1098,12 @@ func (m *multiTestContext) expireLeases() {
 	m.expireLeasesWithoutIncrementingEpochs()
 
 	// Increment epochs.
+	m.mu.RLock()
+	nls := append([]*storage.NodeLiveness(nil), m.nodeLivenesses...)
+	m.mu.RUnlock()
+
 	ctx := context.Background()
-	for idx, nl := range m.nodeLivenesses {
+	for idx, nl := range nls {
 		l, err := nl.Self()
 		if err != nil {
 			continue

